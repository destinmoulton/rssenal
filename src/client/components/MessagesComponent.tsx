import debug from "debug";

import * as React from "react";
import * as NotificationSystem from "react-notification-system";

import { IMessage, TMessages } from "../types";

const log = debug("rssenal:MessagesComponent");

export interface IMessagesMapDispatch {
    messageRemove?: (message: IMessage) => void;
}

export interface IMessagesMapState {
    messages?: TMessages;
}

type TAllProps = IMessagesMapDispatch & IMessagesMapState;

interface IRefs {
    notificationSystem: NotificationSystem.System;
}

class MessagesComponent extends React.Component<TAllProps> {
    private _refs: IRefs = {
        notificationSystem: null
    };

    componentDidUpdate(prevProps: TAllProps) {
        const newMessages = this.props.messages;
        const oldMessages = prevProps.messages;

        newMessages.map((newMessage: IMessage) => {
            if (!oldMessages.includes(newMessage)) {
                newMessage.onRemove = this._onCloseMessage;
                newMessage.position = "br";
                this._refs.notificationSystem.addNotification(newMessage);
            }
        });
    }

    _onCloseMessage = (message: IMessage) => {
        log("_onCloseMessage()");
        this.props.messageRemove(message);
    };

    render() {
        log("render()");
        return (
            <NotificationSystem
                ref={(ref: NotificationSystem.System) =>
                    (this._refs.notificationSystem = ref)
                }
            />
        );
    }
}

export default MessagesComponent;
