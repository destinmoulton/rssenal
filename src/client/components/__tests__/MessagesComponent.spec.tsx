import * as React from "react";
import { shallow } from "enzyme";

import MessagesComponent from "../MessagesComponent";

describe("<MessagesComponent />", () => {
    it("renders and matches snapshot", () => {
        const wrapper = shallow(<MessagesComponent />);

        expect(wrapper).toMatchSnapshot();
    });
});
