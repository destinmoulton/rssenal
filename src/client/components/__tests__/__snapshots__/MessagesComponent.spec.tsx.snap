// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesComponent /> renders and matches snapshot 1`] = `
<Component
  allowHTML={false}
  noAnimation={false}
  style={Object {}}
/>
`;
